src/coreclr/jit/gentree.cpp

```diff
@@ -4805,6 +4813,27 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
 #pragma warning(pop)
 #endif
 
+#ifdef DEBUG
+bool GenTree::OperSupportsReverseOpEvalOrder(Compiler* comp) const
+{
+    if (OperIs(GT_INTRINSIC) && OperIsBinary())
```

Comment:
`GT_INTRINSIC` will always be `OperIsBinary()`, so the check is not quite correct.

You could make it into something like this to share the code for binary operators:
```cs
if (OperIsBinary())
{
    if ((AsOp()->gtGetOp1() == nullptr) || (AsOp()->gtGetOp2() == nullptr))
    {
        return false;
    }

    if (OperIs(GT_COMMA, GT_BOUNDS_CHECK))
    {
        return false;
    }

    if (OperIs(GT_INTRINSIC))
    {
        return !comp->IsIntrinsicImplementedByUserCall(AsIntrinsic()->gtIntrinsicName);
    }
    
    return true;
}
```

Reply:
Thanks, makes sense